tests/utils/syncify.py

```diff
@@ -0,0 +1,19 @@
+import asyncio
+from types import CoroutineType
+from typing import Any, Coroutine, Union
+
+
+def _call(coro: CoroutineType) -> Any:
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        return asyncio.run(coro)
+    else:
+        return loop.run_until_complete(coro)
+
+
+def run_sync(obj: Union[Coroutine[Any, Any, Any], Any]) -> Any:
+    if isinstance(obj, CoroutineType):
+        return _call(obj)
+
+    return obj
```

Comment:
This typing 😢, I might be able to do better?

Reply:
I feel like the only way this get better is with a generic parameter, right? The return type should be a reflection of the incoming function return type. That might actually clean up the input param types as well.

Reply:
Yep, let me see if I can drill this down a bit.

Reply:
Happy to pair on this if it helps. I know introducing generics can sometimes descend into madness

Reply:
What do you think of the updated version? If you think we can do better, down to pair. I'd love for `run_sync`'s signature to be
```
run_sync(obj: Union[Coroutine[Any, Any, ReturnType], ReturnType]) -> ReturnType:
```
but not sure that's feasible.

Reply:
I think that would be great.